docs/architecture-decisions/0002-extend-timdex-ui-for-gis-data.md

```diff
@@ -0,0 +1,96 @@
+environment variable. We've implemented this in many of our Rails apps and found that it's an effective way to manage
+multiple versions of an application (most recently, in TIMDEX API). The downside of using `flipflop` is that it does
+not appear to be especially well maintained, but we're familiar with it and have used it successfully in the past.
+
+### Index detection
+
+Since geospatial records will be in a separate index at first, we can check for the index the app is using (currently
+defined in in an environment variable) to determine which UI elements to display and which type of query to construct.
+This would solve the problem similarly to feature flags. It does not account for the possible future state of geospatial
+data existing in other indices, but it could solve the immediate need of maintaining GDT features separately.
+
+### Search params
+
+Bento uses a `target` param to identify which source is being queried. We could implement a similar design pattern in
+TIMDEX UI to distinguish between geospatial queries and 'general' queries. This would likely include a boolean in the
+search form that would trigger a geospatial query. In addition to providing the query target, we could use the param
+value to evaluate whether to add GIS-specific UI elements to the results view.
+
+Like index detection, this is duplicative of the feature flags approach, but it would allow us to achieve a similar
+effect without using an external dependency. However, this would not be a consistent solution for the full record view,
+as direct linking is a common use case for full records, and we may not be able to guarantee the presence of a `gis=true`
+param in these links.
+
+One use case where search params could be useful is the potential future state in which `search.libraries.mit.edu` (or
+another TIMDEX UI app) includes GIS data in search results. In that instance, it may be valuable to allow users to
+choose whether they want geospatial data queried in their search.
+
+### Data detection
+
+We could check for the presence of certain fields in the API response to determine which UI elements to display. If
+geospatial records are in the results, then GDT-specific fields would appear in the brief and full record, and
+GDT-specific filters would appear in the sidebar. This would help address the use case of adding geospatial data to
+indices searched by non-GDT applications. 
+
+### Theme gem
+
+We can use the theme gem to conditionally modify certain views that are shared across our Rails properties. This would
+solve the problem of customizing the header and possibly the nav, but not views that are specific to the GDT UI.
+
+## Decisions
+
+We will extend TIMDEX UI for GIS data using a combination of the solutions described above:
+
+* We will create a separate Heroku app in the TIMDEX UI pipeline for GDT.
+* Where possible, local header and nav customizations will be controlled by the theme gem.
```

Comment:
I'm comfortable with things like "the theme gem supports an environment variable which implementing sites can use to put their own name in the header", but am nervous about things like "the theme gem has two nav templates, one for GIS and one for RDI. Which one gets loaded is determined by an environment variable".

Reply:
That's fair. I was a bit iffy on managing nav variations from the theme gem, too. I think we can scope this to app names and manage navs in their corresponding codebases.